docs/frontend-system/architecture/08-naming-patterns.md

```diff
@@ -0,0 +1,139 @@
+---
+id: naming-patterns
+title: Frontend System Naming Patterns
+sidebar_label: Naming Patterns
+# prettier-ignore
+description: Naming patterns in the frontend system
+---
+
+> **NOTE: The new frontend system is in a highly experimental phase**
+
+These are the naming patterns to adhere to within the frontend system. They help us keep exports and IDs consistent across packages and make it easier to understand the usage and intent of exports and IDs.
+
+As a rule, all names should be camel case, with the exceptions of plugin and extension IDs, which should use kebab case.
+
+### Plugins
+
+| Description | Pattern      | Examples                              |
+| ----------- | ------------ | ------------------------------------- |
+| ID          | `'<id>'`     | `'catalog'`, `'user-settings'`        |
+| Symbol      | `<id>Plugin` | `catalogPlugin`, `userSettingsPlugin` |
+
+Example:
+
+```ts
+export const userSettingsPlugin = createPlugin({
+  id: 'user-settings',
+  ...
+})
```

Comment:
you could add a clarifying `export { userSettingsPlugin as default }` here, and rewording below to point at that fact, that we do this mostly for illustration and that you could have default-exported the `createPlugin` return value directly if you wanted to.